src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/TokenReplayValidationError.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+
+#nullable enable
+namespace Microsoft.IdentityModel.Tokens
+{
+    internal class TokenReplayValidationError : ValidationError
```

Comment:
this is a very specific error, maybe can follow up offline.

Reply:
Let's touch on this during our sync. We are adding specialised ValidationError subclasses to enforce passing certain custom values that are used to create the exceptions depending on the validation that failed.

Reply:
Can we explore having ValidationError abstract.

Reply:
@jennyf19 we have a goal of providing users with deterministic error results. Currently we are leaning towards providing very verbose information.